### PR TITLE
Fixes for julia 0.7, drop support for 0.5

### PR DESCRIPTION
**Pull request by carlobaldassi**
This fixes all the errors and warnings on Julia 0.7 alpha. Tests pass locally.
Supersedes #67 and #65.

### CODE REVIEW

**Comment by carlobaldassi** [comment]
Update: this now also fixes tests on Windows (#48).

File: .travis.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ os:
     - osx
     - linux
 julia:
-    - 0.5
+    - 0.6
     - nightly
```
Comment (CarloLucibello) [inline .travis.yml]: test on 0.7 as well? (nightly is going to diverge soon)

File: README.md [hunk 2]
```diff
@@ -3,6 +3,7 @@
 [![GZip](http://pkg.julialang.org/badges/GZip_0.3.svg)](http://pkg.julialang.org/?pkg=GZip)
 [![GZip](http://pkg.julialang.org/badges/GZip_0.4.svg)](http://pkg.julialang.org/?pkg=GZip)
 [![GZip](http://pkg.julialang.org/badges/GZip_0.5.svg)](http://pkg.julialang.org/?pkg=GZip)
+[![GZip](http://pkg.julialang.org/badges/GZip_0.6.svg)](http://pkg.julialang.org/?pkg=GZip)
```
Comment (CarloLucibello) [inline README.md]: add 0.7, maybe remove 0.3-0.5?

File: appveyor.yml [hunk 3]
```diff
@@ -1,9 +1,9 @@
 environment:
   matrix:
-  - JULIAVERSION: "julialang/bin/winnt/x86/0.5/julia-0.5-latest-win32.exe"
-  - JULIAVERSION: "julialang/bin/winnt/x64/0.5/julia-0.5-latest-win64.exe"
-  - JULIAVERSION: "julianightlies/bin/winnt/x86/julia-latest-win32.exe"
-  - JULIAVERSION: "julianightlies/bin/winnt/x64/julia-latest-win64.exe"
+  - JULIA_URL: "https://julialang-s3.julialang.org/bin/winnt/x86/0.6/julia-0.6-latest-win32.exe"
+  - JULIA_URL: "https://julialang-s3.julialang.org/bin/winnt/x64/0.6/julia-0.6-latest-win64.exe"
```
Comment (CarloLucibello) [inline appveyor.yml]: + 0.7

File: .travis.yml (hunk 1 above)
Comment (carlobaldassi) [reply .travis.yml]: Sure, if 0.7 binaries are available on Travis already. Are they?

File: README.md (hunk 2 above)
Comment (carlobaldassi) [reply README.md]: Will add 0.7 as soon as I have my laptop back (hopefully in a couple of days).

File: appveyor.yml (hunk 3 above)
Comment (carlobaldassi) [reply appveyor.yml]: same as above, will do soon-ish. In the meanwhile this PR could be merged as is, these changes can be applied afterwards.

File: .travis.yml (hunk 1 above)
Comment (KristofferC) [reply .travis.yml]: Yes

**Comment by CarloLucibello** [comment]
@ararslan @KristofferC someone could merge this PR?

File: src/GZip.jl [hunk 4]
```diff
@@ -166,39 +172,39 @@ end
 
 # Easy access to gz reading/writing functions (Internal)
 gzgetc(s::GZipStream) =
-    @test_eof_gzerr(s, ccall((:gzgetc, _zlib), Int32, (Ptr{Void},), s.gz_file), -1)
+    @test_eof_gzerr(s, ccall((:gzgetc, _zlib), Int32, (Ptr{Nothing},), s.gz_file), -1)
```
Comment (ararslan) [inline src/GZip.jl]: We have `Cvoid` for this purpose. (It's an alias for `Nothing`, but it's intended to be used in `ccall` for clarity.)

File: src/GZip.jl [hunk 5]
```diff
@@ -80,23 +83,26 @@ const GZLIB_VERSION = unsafe_string(ccall((:zlibVersion, GZip._zlib), Ptr{UInt8}
...
+    gz_file::Ptr{Nothing}
     buf_size::Int
 
     _closed::Bool
 
-    GZipStream(name::AbstractString, gz_file::Ptr{Void}, buf_size::Int) =
-        (x = new(name, gz_file, buf_size, false); finalizer(x, close); x)
+    function GZipStream(name::AbstractString, gz_file::Ptr{Nothing}, buf_size::Int)
+        x = new(name, gz_file, buf_size, false)
+        VERSION ≥ v"0.7.0-DEV.2562" ? finalizer(close, x) : finalizer(x, close)
```
Comment (ararslan) [inline src/GZip.jl]: `@compat finalizer(close, x)`

**Comment by carlobaldassi** [comment]
Ok I have addressed all the issues. Ready to merge?

**Comment by carlobaldassi** [comment]
Thanks @ararslan. #65 and #67 should be closed now, since they were superseded by this PR.

**Pull request merged**